packages/react/src/createReactQueryHooks.tsx

```diff
@@ -60,6 +61,31 @@ interface UseTRPCMutationOptions<TInput, TError, TOutput>
   extends UseMutationOptions<TOutput, TError, TInput>,
     TRPCUseQueryBaseOptions {}
 
+type Override<A, B> = {
+  [K in keyof A]: K extends keyof B ? B[K] : A[K];
+};
+type Equals<X, Y> = (<T>() => T extends X ? 1 : 2) extends <T>() => T extends Y
+  ? 1
+  : 2
+  ? true
+  : false;
+
+type UseTRPCMutationResult<TInput, TOutput, TError, TOptions> = Equals<
```

Comment:
Is `Equals` definitely needed here? It seems quite big/complex. Maybe this would work:

```ts
[NonNullable<TInput>] extends [never]
  ? ... nullish...
  : ...not nullish...

Reply:
I agree that Equals is a bit complex but for example if input is `z.string().nullish()` that means `TInput` would `string | null | undefined`, calling `NonNullable<TInput>` would just return `string`.

`Equals<TInput, NonNullable<TInput>>` seemed like the simplest way, and checking if `TInput extends null | undefined` narrows the type which results in `TInput` being useless for the nullish branch of the ternary.

also Equals looks worse than it actually is because of prettier formatting, it should actually look this https://github.com/Microsoft/TypeScript/issues/27024#issuecomment-421529650